src/main/extension/extension_load.cpp

```diff
@@ -207,6 +207,22 @@ bool ExtensionHelper::TryInitialLoad(DBConfig &config, FileSystem &fs, const str
 		throw IOException("Extension \"%s\" could not be loaded: %s", filename, GetDLError());
 	}
 
+	ext_version_fun_t error_fun = nullptr;
+	try {
+		error_fun = LoadFunctionFromDLL<ext_version_fun_t>(lib_hdl, "error_message", filename);
+	} catch (...) {
+		// If error_message function is not present, move further
+	}
+
+	if (error_fun) {
+		auto error_message = (*error_fun)();
+		if (error_message) {
+			throw InvalidInputException(
+			    "Extension \"%s\" exposes an error_message function\nSelf-reported error message: \"%s\"", filename,
```

Comment:
```suggestion
			    "Loading extension \"%s\" failed: \"%s\"", filename,
```

I'm not sure the user really cares *how* the exception was thrown?

Reply:
Very open to suggestions on how to properly say this, but I think ideally it would need to convey that one part of the message comes from the extension developer (so has can have a different register, can even be random like "Quack!") and not from DuckDB.

Reply:
See https://github.com/duckdb/duckdb/pull/8834#issuecomment-1712254096